src/libraries/Common/tests/System/TimeProviderTests.cs

```diff
@@ -638,6 +638,53 @@ public override ITimer CreateTimer(TimerCallback callback, object? state, TimeSp
             }
         }
 
+        public static IEnumerable<object[]> TaskFactoryData()
+        {
+            yield return new object[] { taskFactory };
+
+#if TESTEXTENSIONS
+            yield return new object[] { extensionsTaskFactory };
+#endif // TESTEXTENSIONS
+        }
+
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        [MemberData(nameof(TaskFactoryData))]
+        public void TestDelayTaskContinuation(ITestTaskFactory taskFactory)
+        {
+            //
+            // Test time expiration and validate continuation is called synchronously.
+            //
+
+            ManualTimeProvider manualTimeProvider = new ManualTimeProvider();
+            int callbackCount = 0;
+
+            _ = Continuation(manualTimeProvider, default, () => callbackCount++);
+
+             Assert.NotNull(manualTimeProvider.Timer);
+             manualTimeProvider.Timer.Fire();
+
+            // Delay should be completed and the continuation should be called synchronously.
```

Comment:
I typically validate synchronous vs asynchronous callbacks by using threadlocals. For example, if you do:
```C#
var tl = new ThreadLocal<int>();
tl.Value = 42;
int readValue = 0;
InvokeCallback(() => readValue = tl.Value);
Assert.Equal(42, readValue);
```
the only way that should pass is if the callback was invoked synchronously by the current thread.  Conversely for asynchronous, you can validate that the value is not equal to what it was set to.